source/Heuristics.h

```diff
@@ -10,122 +10,183 @@
 #include <cstdint>
 #include <optional>
 
+#include <json/json.h>
+
 namespace marianatrench {
 
-class Heuristics {
- public:
+class Heuristics final {
+public:
+  explicit Heuristics(
+      std::size_t join_override_threshold = join_override_threshold_default,
+      std::size_t android_join_override_threshold =
+          android_join_override_threshold_default,
+      std::optional<std::size_t> warn_override_threshold =
+          warn_override_threshold_default,
+      std::size_t generation_max_port_size = generation_max_port_size_default,
+      std::size_t generation_max_output_path_leaves =
+          generation_max_output_path_leaves_default,
+      std::size_t parameter_source_max_port_size =
+          parameter_source_max_port_size_default,
+      std::size_t parameter_source_max_output_path_leaves =
+          parameter_source_max_output_path_leaves_default,
+      std::size_t sink_max_port_size = sink_max_port_size_default,
+      std::size_t sink_max_input_path_leaves =
+          sink_max_input_path_leaves_default,
+      std::size_t call_effect_source_max_port_size =
+          call_effect_source_max_port_size_default,
+      std::size_t call_effect_source_max_output_path_leaves =
+          call_effect_source_max_output_path_leaves_default,
+      std::size_t call_effect_sink_max_port_size =
+          call_effect_sink_max_port_size_default,
+      std::size_t call_effect_sink_max_input_path_leaves =
+          call_effect_sink_max_input_path_leaves_default,
+      std::size_t max_number_iterations = max_number_iterations_default,
+      std::size_t max_depth_class_properties =
+          max_depth_class_properties_default,
+      std::size_t max_call_chain_source_sink_distance =
+          max_call_chain_source_sink_distance_default,
+      std::size_t propagation_max_input_path_size =
+          propagation_max_input_path_size_default,
+      std::size_t propagation_max_input_path_leaves =
+          propagation_max_input_path_leaves_default);
+
+  bool operator==(const Heuristics &other) const;
+
+  static Heuristics from_json(const Json::Value &value);
+
   /**
    * When a method has a set of overrides greater than this threshold, we do not
    * join all overrides at call sites.
    */
-  constexpr static std::size_t kJoinOverrideThreshold = 40;
+  std::size_t join_override_threshold() const;
```

Comment:
ulta nit: I would inline the code of all these getters in the .h, so the compiler can inline them.
EDIT: To be fair, `Options` also has a lot of getter methods that are not inlined. so we could keep it as-is for consistency, I guess.